File: samples/wordnik-api-scala/src/main/scala/com/wordnik/client/model/WordOfTheDay.scala
package com.wordnik.client.model

import java.util.Date
import scala.reflect.BeanProperty

class WordOfTheDay {
  @BeanProperty var id: Long = 0L
  @BeanProperty var parentId: String = _
  @BeanProperty var category: String = _
  @BeanProperty var createdBy: String = _
  @BeanProperty var createdAt: Date = _
  @BeanProperty var contentProvider: ContentProvider = _
  @BeanProperty var word: String = _
  @BeanProperty var htmlExtra: String = _
  @BeanProperty var definitions: java.util.List[Definition] = new java.util.ArrayList[Definition]()
  @BeanProperty var examples: java.util.List[Example] = new java.util.ArrayList[Example]()
  @BeanProperty var publishDate: Date = _
  @BeanProperty var note: String = _
  override def toString: String = {
    val sb = new StringBuilder
    sb.append("class WordOfTheDay {\n")
    sb.append("  id: ").append(id).append("\n")
    sb.append("  parentId: ").append(parentId).append("\n")
    sb.append("  category: ").append(category).append("\n")
    sb.append("  createdBy: ").append(createdBy).append("\n")
    sb.append("  createdAt: ").append(createdAt).append("\n")
    sb.append("  contentProvider: ").append(contentProvider).append("\n")
    sb.append("  word: ").append(word).append("\n")
    sb.append("  htmlExtra: ").append(htmlExtra).append("\n")
    sb.append("  definitions: ").append(definitions).append("\n")
    sb.append("  examples: ").append(examples).append("\n")
    sb.append("  publishDate: ").append(publishDate).append("\n")
    sb.append("  note: ").append(note).append("\n")
    sb.append("}\n")
    sb.toString
  }
}

File: samples/wordnik-api-ruby/models/WordOfTheDay.rb
class WordOfTheDay
  attr_accessor :id, :parent_id, :category, :created_by, :created_at, :content_provider, :word, :html_extra, :definitions, :examples, :publish_date, :note

  # :internal => :external
  def self.attribute_map
  {
      :id => :id, :parent_id => :parentId, :category => :category, :created_by => :createdBy, :created_at => :createdAt, :content_provider => :contentProvider, :word => :word, :html_extra => :htmlExtra, :definitions => :definitions, :examples => :examples, :publish_date => :publishDate, :note => :note

  }
  end

  def initialize(attributes = {})
    # Morph attribute keys into undescored rubyish style
    if attributes.to_s != ""
      WordOfTheDay.attribute_map.each_pair do |internal, external|
        if attributes.has_key? external
          attributes[internal] = attributes.delete(external)
        end
      end

      # Assign attributes
      attributes.each do |name, value|
        send("#{name}=", value) if self.respond_to?(name.to_sym)
      end
    end
  end

  def to_body
    body = {}
    WordOfTheDay.attribute_map.each_pair do |key,value|
      body[value] = self.send(key) unless self.send(key).nil?
    end
    body
  end
end
